File: src/mysqlcookbook/recipes/api/05_null_in_result/null-in-result.php
<?php
# null-in-result.php: print undef values as "NULL"

require_once "Cookbook.php";

try
{
  $dbh = Cookbook::connect ();
}
catch (PDOException $e)
{
  print ("Error message: " . $e->getMessage () . "\n");
  exit (1);
}

try
{
#@ _FETCHLOOP_1_
  $sth = $dbh->query ("SELECT name, birth, foods FROM profile");
  while ($row = $sth->fetch (PDO::FETCH_NUM))
  {
    foreach (array_keys ($row) as $key)
    {
      if ($row[$key] === NULL)
        $row[$key] = "NULL";
    }
    print ("name: $row[0], birth: $row[1], foods: $row[2]\n");
  }
#@ _FETCHLOOP_1_
}
catch (PDOException $e)
{
  print ("Oops, the statement failed\n");
  print ("Error message: " . $e->getMessage () . "\n");
}

$dbh = NULL;
?>

File: src/mysqlcookbook/recipes/api/05_null_in_result/null-in-result.rb
#!/usr/bin/ruby
# null-in-result.rb: demonstrate NULL values in the result set
# processing in Ruby

require "Cookbook"

begin
  client = Cookbook.connect
rescue Mysql2::Error => e
  puts "Cannot connect to server"
  puts "Error code: #{e.errno}"
  puts "Error message: #{e.message}"
  exit(1)
end

begin
#@ _EACHLOOP_
  result = client.query("SELECT name, birth, foods FROM profile")
  result.each do |row|
    printf "name %s, birth: %s, foods: %s\n", 
           row["name"].nil? ? "NULL" : row["name"], 
           row["birth"].nil? ? "NULL" : row["birth"], 
           row["foods"].nil? ? "NULL" : row["foods"]
  end
#@ _EACHLOOP_
rescue Mysql2::Error => e
  puts "Oops, the statement failed"
  puts "Error code: #{e.errno}"
  puts "Error message: #{e.message}"
end

client.close
